base/util.jl

```diff
@@ -657,6 +657,40 @@ function def_name_defval_from_kwdef_fielddef(kwdef)
     end
 end
 
+
+function typeconst_ex(ex)
+    ex isa Expr || return ex
+    if ex.head === :(=)
+        quote
+            tmp = $(esc(ex.args[2]))
```

Comment:
As I wrote in https://giordano.github.io/blog/2022-06-18-first-macro/, you want to mark `tmp` as local to avoid leaking the name to the calling namespace
```suggestion
            local tmp = $(esc(ex.args[2]))
```

Reply:
Thanks! I figured it was OK because it was gensymed (I had a previous version that called gensym() directly), but you're right it's even better not to leak it at all!